sky/serve/controller.py

```diff
@@ -142,6 +175,44 @@ async def update_service(request: fastapi.Request):
                              f'{common_utils.format_exception(e)}')
                 return {'message': 'Error'}
 
+        @self._app.post('/controller/terminate_replica')
+        async def terminate_replica(request: fastapi.Request):
+            request_data = await request.json()
+            try:
+                replica_id = request_data.get('replica_id')
+                if replica_id is None:
+                    return {
+                        'code': 400,
+                        'message': 'Error: replica ID is not specified.'
+                    }
+                purge = request_data.get('purge')
+                if purge is None:
+                    return {
+                        'code': 400,
+                        'message': 'Error: purge is not specified.'
+                    }
```

Comment:
We can assert those vars is not None as this endpoint is only accessed by our code?

Reply:
saw the issue you filed. Good point, lets keep it in a separate PR.